src/main/java/io/pakland/mdas/githubstats/application/FetchRepositoriesFromTeam.java

```diff
@@ -0,0 +1,20 @@
+package io.pakland.mdas.githubstats.application;
+
+import io.pakland.mdas.githubstats.application.dto.RepositoryDTO;
+import io.pakland.mdas.githubstats.application.exceptions.HttpException;
+import io.pakland.mdas.githubstats.domain.repository.RepositoryExternalRepository;
+
+import java.util.List;
+
+public class FetchRepositoriesFromTeam {
+
+    private final RepositoryExternalRepository repositoryExternalRepository;
+
+    public FetchRepositoriesFromTeam(RepositoryExternalRepository repositoryExternalRepository) {
+        this.repositoryExternalRepository = repositoryExternalRepository;
+    }
+
+    public List<RepositoryDTO> execute(Integer organizationId, Integer teamId) throws HttpException {
+        return this.repositoryExternalRepository.fetchTeamRepositories(organizationId, teamId);
+    }
```

Comment:
I was thinking on also doing the mapping from DTO to the entity here. From my point of view, we should delegate the responsibility of doing so to the UC instead of the controller.  
Another reason to do so, is that the controller will most likely want to work with the entities, instead of the dtos. Thoughts?

Something like this:

```suggestion
    public List<Repository> execute(Integer organizationId, Integer teamId) throws  public List<Repository> execute(Integer orgId, Integer teamId) throws HttpException {
       List<RepositoryDTO> repositoryDTOS = this.repositoryExternalRepository.fetchTeamRepositories(
           orgId, teamId
       );
       return repositoryDTOS.stream().map(RepositoryMapper::dtoToEntity).toList();
    }
    }
```

Reply:
I'm with you, we should definitely avoid doing the mappings on the controllers. 
We could either work with entities on the controller or keep the use of entities only for the domain and application layers and therefore use exclusively DTOs on the controller. The second option would imply doing the mappings only on the usecases which interact with the database repositories.

Reply:
Let's use DTO's in controllers and making the UC that will save the DTO's to the database do the mapping then. 👌🏼